src/pygeoprocessing/geoprocessing_core.pyx

```diff
@@ -656,22 +656,20 @@ def raster_band_percentile(
         will select the next element higher than the percentile cutoff).
 
     """
-    raster_type = pygeoprocessing.get_raster_info(
-        base_raster_path_band[0])['datatype']
-    if raster_type in (
-            gdal.GDT_Byte, gdal.GDT_Int16, gdal.GDT_UInt16, gdal.GDT_Int32,
-            gdal.GDT_UInt32):
+    numpy_type = pygeoprocessing.get_raster_info(
+        base_raster_path_band[0])['numpy_type']
+    if numpy.issubdtype(numpy_type, numpy.integer):
```

Comment:
This is a really nice update - one less thing to have to worry about updating if GDAL adds more int or float types!